packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts

```diff
@@ -56,58 +56,75 @@ describe('fetch httpClient testing', () => {
   });
 
   it('cancels equal endpoint requests if no requestId parameter is passed', async () => {
-    expect.assertions(2);
     await Promise.all([
-      fetchHttpClient(endpoint, {
-        parameters: {
-          q: 'shirt'
-        }
-      }).catch(error => expect(error.code).toEqual(DOMException.ABORT_ERR)),
-      fetchHttpClient(endpoint, {
-        parameters: {
-          q: 'jeans'
-        }
-      }).then(response => expect(response).toBeDefined())
+      expect(
+        fetchHttpClient(endpoint, {
+          parameters: {
+            q: 'shirt'
+          }
+        })
+      ).rejects.toHaveProperty('code', DOMException.ABORT_ERR),
+      expect(
+        fetchHttpClient(endpoint, {
+          parameters: {
+            q: 'jeans'
+          }
+        })
+      ).resolves.toBeDefined()
     ]);
   });
 
+  // eslint-disable-next-line max-len
+  it('does not cancel equal endpoint requests if a cancelable=false parameter is passed', async () => {
+    expect(
+      await Promise.all([
+        fetchHttpClient(endpoint, {
+          cancelable: false,
+          parameters: {
+            q: 'shirt'
+          }
+        }),
+        fetchHttpClient(endpoint, {
+          parameters: {
+            q: 'jeans'
+          }
+        })
+      ])
+    ).toHaveLength(2);
+  });
+
   // eslint-disable-next-line max-len
   it('does not cancel equal endpoint requests if a different requestId parameter is passed', async () => {
-    expect.assertions(3);
-    const searchUrl = `${endpoint}?additionalParam=true`;
-    const empathizeUrl = 'https://api.empathy.co/empathize?additionalParam=true';
-    await Promise.all([
-      fetchHttpClient(searchUrl, {
-        id: 'search',
-        parameters: {
-          q: 'shirt'
-        }
-      }).catch(error => expect(error.code).toEqual(DOMException.ABORT_ERR)),
-      fetchHttpClient(empathizeUrl, {
-        id: 'search',
-        parameters: {
-          q: 'shirt'
-        }
-      }).then(response => expect(response).toBeDefined()),
-      fetchHttpClient(searchUrl, {
-        parameters: {
-          q: 'jeans'
-        }
-      }).then(response => expect(response).toBeDefined())
-    ]);
+    expect(
```

Comment:
Shouldn't this test be done with cancellable param to true and using different IDs for the requests?

Reply:
![image](https://user-images.githubusercontent.com/75546736/186392596-f15828e1-ee16-40c7-a905-7ee424cc2f47.png)